tests/tests_pytorch/models/test_restore.py

```diff
@@ -255,13 +255,36 @@ class TestModel(BoringModel):
         def on_train_start(self) -> None:
             assert self.trainer.current_epoch == first_max_epochs
             assert self.trainer.global_step == first_max_epochs * train_batches
+            assert self.trainer.fit_loop.epoch_loop._batches_that_stepped == first_max_epochs * train_batches
 
     trainer.fit(TestModel(), ckpt_path=ckpt_path)
     assert trainer.current_epoch == max_epochs
     assert trainer.global_step == max_epochs * train_batches
     assert trainer.fit_loop.epoch_loop._batches_that_stepped == max_epochs * train_batches
 
 
+def test_logging_step_loaded_correctly_pre_1_6_5(tmpdir):
```

Comment:
Have you considered writing a unit test instead? If so, why is it not possible?

Reply:
because this is how we restore global_step
https://github.com/Lightning-AI/lightning/blob/208c85013528704f4cc9727d44c95ebf5cde0b34/src/pytorch_lightning/trainer/connectors/checkpoint_connector.py#L270

and by this point, the optimizer progress is not loaded, so it's not possible to assign the global step if we go with unit test approach.

Reply:
oh wait! just realized it might not work with manual optimization.